contracts/auto-withdrawer/src/contract.rs

```diff
@@ -226,33 +266,123 @@ pub fn reply(
             CORE_UNBOND.remove(deps.storage);
             // it is safe to use unwrap() here since this reply is only called on success
             let events = reply.result.unwrap().events;
+            deps.api.debug(&format!("WASMDEBUG: {:?}", events));
             reply_core_unbond(deps, sender, deposit, events)
         }
-        _ => unreachable!(),
+        id => Err(ContractError::InvalidCoreReplyId { id }),
```

Comment:
I still that think that the use of unreachable!() macro was more reasonable here. Saves code space (and consenquantially gas) and reduces amount of code for something that would never happen anyway.

Reply:
cause, as I understand it, the reply method is one of the external handlers of the contract, and it can be called separately from the external environment with incorrect data

Reply:
`reply()` is called by Neutron blockchain using the reply ID which we pass into it during the very same transaction. If Neutron binary for some reason decides to screw us and use a malformed reply ID, then we are in a big trouble regardless of what we do. That said, I don't see any value in introducing error checks in contexts outside of our scope.

Reply:
Fixed